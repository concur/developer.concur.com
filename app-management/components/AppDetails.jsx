/* eslint-env browser */

import React, { PropTypes } from 'react';
import { Link } from 'react-router';

class AppDetails extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      app: {},
      loading: false,
      error: false,
    };
  }

  componentWillMount() {
    window.fetch(`${process.env.API_SERVER}/apps/${this.props.params.id}`)
      .then(response => response.json())
      .then(app => this.setState({
        app,
        loading: false,
      }))
      .catch(() => this.setState({
        error: true,
        loading: false,
      }));
  }

  render () {
    if (this.state.loading) {
      return (
        <div className="row">
          <div className="col-md-12">
            <p>Loading...</p>
          </div>
        </div>
      );
    }

    if (this.state.error) {
      return (
        <div className="row">
          <div className="col-md-12">
            <p>An error occurred when loading this app</p>
            <Link to="/">Back to My Apps</Link>
          </div>
        </div>
      );
    }

    const { app } = this.state;

    return (
      <div className="row">
        <div className="col-md-12">
          <h2>{app.name}</h2>
          <Link to="/">&larr; Back to My Apps</Link>
          <div className="row">
            <p>{app.description}</p>
          </div>
        </div>
      </div>
    );
  }
}

AppDetails.propTypes = {
  params: PropTypes.shape({
    id: PropTypes.string.isRequired,
  }),
};

export default AppDetails;
