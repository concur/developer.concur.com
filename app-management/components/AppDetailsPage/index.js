/* eslint-env browser */

import { connect } from 'react-redux';
import AppDetailsPage from './AppDetailsPage';
import { fetchAppDetails, updateAppDetails, showSecret } from '../../actions/appDetails';

export function mapStateToProps(state) {
  return state.appDetails;
}

export function mapDispatchToProps(dispatch) {
  return {
    fetchAppDetails: id => dispatch(fetchAppDetails(id)),
    handleSubmit: app => dispatch(updateAppDetails(app)),
    generateSecret: (id) => {
      const confirmation = window.confirm('Are you sure? This will clear your previous app secret.');
      if (confirmation) {
        dispatch(showSecret());
        console.log(`TBD: Generate secret for app ${id}`);
      }
    },
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AppDetailsPage);
