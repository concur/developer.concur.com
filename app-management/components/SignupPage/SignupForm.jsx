import React, { PropTypes } from 'react';
import { Field, reduxForm } from 'redux-form';

import { formValidator } from '../../utils/formValidator';
import { InputField, SelectField, CheckboxField } from '../FormFields';

import countries from '../../data/countries.json';
import concurRelationships from '../../data/concurRelationships.json';

const constraints = {
  firstName: {
    presence: { message: 'is required' },
  },
  lastName: {
    presence: { message: 'is required' },
  },
  company: {
    presence: { message: 'is required' },
  },
  email: {
    presence: { message: 'is required' },
    email: true,
  },
  password: {
    presence: { message: 'is required' },
    length: { minimum: 8 },
  },
  passwordVerify: {
    presence: { message: 'is required' },
    equality: {
      attribute: 'password',
      message: 'must match password',
    },
  },
  country: {
    presence: { message: 'is required' },
  },
  website: {
    url: true,
  },
  termsOfUseAgreement: {
    presence: { message: 'is required' },
    inclusion: {
      within: [true],
      message: 'is required',
    },
  },
};

const SignupForm = ({ handleSubmit }) => (
  <form onSubmit={handleSubmit}>
    <fieldset>
      <div className="row">
        <section className="col-md-6">
          <Field
            component={InputField}
            type="text"
            name="firstName"
            label="First Name &#42;"
            placeholder="First Name"
          />
        </section>
        <section className="col-md-6">
          <Field
            component={InputField}
            type="text"
            name="lastName"
            label="Last Name &#42;"
            placeholder="Last Name"
          />
        </section>
      </div>
      <div className="row">
        <section className="col-md-6">
          <Field
            component={InputField}
            type="email"
            name="email"
            label="Email Address &#42;"
            placeholder="john.smith@example.com"
          />
        </section>
        <section className="col-md-6">
          <Field
            component={InputField}
            type="text"
            name="company"
            label="Company Name &#42;"
            placeholder="Company Name"
          />
        </section>
      </div>
      <div className="row">
        <section className="col-md-6">
          <Field
            component={InputField}
            type="password"
            name="password"
            label="Password (8 characters minimum) &#42;"
            placeholder="Password"
          />
        </section>
        <section className="col-md-6">
          <Field
            component={InputField}
            type="password"
            name="passwordVerify"
            label="Password Verify &#42;"
            placeholder="Password Verify"
          />
        </section>
      </div>
      <div className="row">
        <section className="col-md-6">
          <Field
            component={InputField}
            type="text"
            name="businessDescription"
            label="Business Description"
            placeholder="Description"
          />
        </section>
        <section className="col-md-6">
          <Field
            component={InputField}
            type="tel"
            name="phoneNumber"
            label="Phone Number"
            placeholder="+1-555-555-5555"
          />
        </section>
      </div>
      <div className="row">
        <section className="col-md-6">
          <Field
            component={SelectField}
            name="country"
            label="Country &#42;"
            placeholder="Select country"
            options={countries}
            simpleValue
          />
        </section>
        <section className="col-md-6">
          <Field
            component={InputField}
            type="url"
            name="website"
            label="Website"
            placeholder="http://example.com"
          />
        </section>
      </div>
      <div className="row">
        <section className="col-md-6">
          <Field
            component={SelectField}
            name="relationshipToConcur"
            label="Relationship with Concur"
            placeholder="Select relationship"
            options={concurRelationships}
            simpleValue
          />
        </section>
      </div>
      <section>
        <Field
          component={CheckboxField}
          name="termsOfUseAgreement"
        >
          I Agree to the <a href="/Terms-of-Use.html" target="_blank" rel="noopener noreferrer">Terms of Use</a> and <a href="https://www.concur.com/en-us/privacy-policy" target="_blank" rel="noopener noreferrer">Privacy Policy</a>
        </Field>
      </section>
      <p>&#42; required field</p>
    </fieldset>
    <footer>
      <button type="submit" className="btn bright-blue pull-right">Sign Up</button>
    </footer>
  </form>
);

SignupForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
};

export default reduxForm({
  form: 'signup',
  validate: formValidator(constraints),
})(SignupForm);
