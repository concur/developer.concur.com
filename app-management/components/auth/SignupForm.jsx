import React from 'react';

class SignupForm extends React.Component {
  constructor(props) {
    super(props);

    this.submitHandler = this.submitHandler.bind(this);
  }

  submitHandler(e) {
    e.preventDefault();
  }

  render () {
    return (
      <div className="row">
        <div className="col-md-12">
          <h2>Signup</h2>
          <form
            className="sky-form"
            method="POST"
            action={`${process.env.API_SERVER}/signup`}
            onSubmit={this.submitHandler}
          >
            <fieldset>
              <div className="row">
                <section className="col col-6">
                  <label className="label" htmlFor="firstName">First Name</label>
                  <div className="input">
                    <input type="text" name="firstName" id="firstName" placeholder="First Name" />
                  </div>
                </section>
                <section className="col col-6">
                  <label className="label" htmlFor="lastName">Last Name</label>
                  <div className="input">
                    <input type="text" name="lastName" id="lastName" placeholder="Last Name" />
                  </div>
                </section>
              </div>
              <div className="row">
                <section className="col col-6">
                  <label className="label" htmlFor="username">Username</label>
                  <div className="input">
                    <input type="text" name="username" id="username" placeholder="Username" />
                  </div>
                </section>
                <section className="col col-6">
                  <label className="label" htmlFor="password">Email Address</label>
                  <div className="input">
                    <input type="email" name="email" id="email" placeholder="Email Address" />
                  </div>
                </section>
              </div>
              <div className="row">
                <section className="col col-6">
                  <label className="label" htmlFor="password">Password</label>
                  <div className="input">
                    <input type="password" name="password" id="password" placeholder="Password" />
                  </div>
                </section>
                <section className="col col-6">
                  <label className="label" htmlFor="passwordConfirm">Confirm Password</label>
                  <div className="input">
                    <input
                      type="password"
                      name="passwordConfirm"
                      id="passwordConfirm"
                      placeholder="Confirm Password"
                    />
                  </div>
                </section>
              </div>
            </fieldset>
            <footer>
              <button type="submit" className="button">Sign Up</button>
            </footer>
          </form>
        </div>
      </div>
    );
  }
}

export default SignupForm;
