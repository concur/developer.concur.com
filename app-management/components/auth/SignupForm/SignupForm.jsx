import React, { PropTypes } from 'react';

class SignupForm extends React.Component {
  static submitHandler(e) {
    e.preventDefault();
  }

  render () {
    const {
      firstName,
      lastName,
      username,
      email,
      password,
      passwordConfirm,
      handleInputChange } = this.props;

    return (
      <div className="row">
        <div className="col-md-12">
          <h2>Signup</h2>
          <form
            className="sky-form"
            method="POST"
            action={`${process.env.API_SERVER}/signup`}
            onSubmit={this.submitHandler}
          >
            <fieldset>
              <div className="row">
                <section className="col col-6">
                  <label className="label" htmlFor="firstName">First Name</label>
                  <div className="input">
                    <input
                      type="text"
                      name="firstName"
                      id="firstName"
                      placeholder="First Name"
                      value={firstName}
                      onChange={handleInputChange}
                    />
                  </div>
                </section>
                <section className="col col-6">
                  <label className="label" htmlFor="lastName">Last Name</label>
                  <div className="input">
                    <input
                      type="text"
                      name="lastName"
                      id="lastName"
                      placeholder="Last Name"
                      value={lastName}
                      onChange={handleInputChange}
                    />
                  </div>
                </section>
              </div>
              <div className="row">
                <section className="col col-6">
                  <label className="label" htmlFor="username">Username</label>
                  <div className="input">
                    <input
                      type="text"
                      name="username"
                      id="username"
                      placeholder="Username"
                      value={username}
                      onChange={handleInputChange}
                    />
                  </div>
                </section>
                <section className="col col-6">
                  <label className="label" htmlFor="password">Email Address</label>
                  <div className="input">
                    <input
                      type="email"
                      name="email"
                      id="email"
                      placeholder="Email Address"
                      value={email}
                      onChange={handleInputChange}
                    />
                  </div>
                </section>
              </div>
              <div className="row">
                <section className="col col-6">
                  <label className="label" htmlFor="password">Password</label>
                  <div className="input">
                    <input
                      type="password"
                      name="password"
                      id="password"
                      placeholder="Password"
                      value={password}
                      onChange={handleInputChange}
                    />
                  </div>
                </section>
                <section className="col col-6">
                  <label className="label" htmlFor="passwordConfirm">Confirm Password</label>
                  <div className="input">
                    <input
                      type="password"
                      name="passwordConfirm"
                      id="passwordConfirm"
                      placeholder="Confirm Password"
                      value={passwordConfirm}
                      onChange={handleInputChange}
                    />
                  </div>
                </section>
              </div>
            </fieldset>
            <footer>
              <button type="submit" className="button">Sign Up</button>
            </footer>
          </form>
        </div>
      </div>
    );
  }
}

SignupForm.propTypes = {
  handleInputChange: PropTypes.func.isRequired,
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  passwordConfirm: PropTypes.string.isRequired,
};

export default SignupForm;
