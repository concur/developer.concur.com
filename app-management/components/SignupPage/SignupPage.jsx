import React, { PropTypes } from 'react';
import SignupForm from './SignupForm';
import LoadingSpinner from '../LoadingSpinner';
import ErrorAlert from '../ErrorAlert';

const SignupPage = ({ handleSubmit, isFetching, error }) => (
  <div className="row">
    <div className="col-md-12">
      <h2>Signup</h2>
      <LoadingSpinner loading={isFetching} />
      <ErrorAlert error={error} />
      <SignupForm onSubmit={handleSubmit} />
    </div>
  </div>
);

SignupPage.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  isFetching: PropTypes.bool,
  error: PropTypes.string,
};

export default SignupPage;
