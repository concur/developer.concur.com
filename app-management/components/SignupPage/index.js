import { connect } from 'react-redux';
import SignupPage from './SignupPage';
import { postSignup } from '../../actions/signup';

export function mapStateToProps(state) {
  return state.signup;
}

export function mapDispatchToProps(dispatch) {
  return {
    handleSubmit: user => dispatch(postSignup(user)),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SignupPage);
