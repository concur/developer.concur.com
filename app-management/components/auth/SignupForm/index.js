import { connect } from 'react-redux';
import SignupForm from './SignupForm';
import { handleInputChange } from '../../../actions/signup';

function mapStateToProps(state) {
  return state.signupForm;
}

function mapDispatchToProps(dispatch) {
  return {
    handleInputChange: e => dispatch(handleInputChange(e.target.name, e.target.value)),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SignupForm);
