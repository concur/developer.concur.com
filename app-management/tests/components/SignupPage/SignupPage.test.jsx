import React from 'react';
import { shallow } from 'enzyme';
import { createStore } from 'redux';
import { shallowToJson } from 'enzyme-to-json';

import SignupPage from '../../../components/SignupPage/SignupPage';
import { mapStateToProps, mapDispatchToProps } from '../../../components/SignupPage';
import SignupForm from '../../../components/SignupPage/SignupForm';
import appReducer from '../../../reducers';

const store = createStore(appReducer);

describe('<SignupPage />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render with <SignupForm />', () => {
    const handleSubmit = jest.fn();
    const page = shallow(<SignupPage handleSubmit={handleSubmit} />);

    expect(shallowToJson(page)).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('returns entire signup state', () => {
      const state = mapStateToProps(store.getState());

      expect(state).toEqual(store.getState().signup);
    });
  });

  describe('mapDispatchToProps', () => {
    it('returns submit handler', () => {
      const dispatcher = jest.fn();
      const state = mapDispatchToProps(dispatcher);
      state.handleSubmit('user');

      expect(state.handleSubmit).toBeDefined();
      expect(dispatcher).toHaveBeenCalled();
    });
  });
});
