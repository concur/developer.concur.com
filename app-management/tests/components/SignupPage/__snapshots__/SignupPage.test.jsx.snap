exports[`<SignupPage /> should render with <SignupForm /> 1`] = `
<div
  className="row">
  <div
    className="col-md-12">
    <h2>
      Signup
    </h2>
    <LoadingSpinner />
    <ErrorAlert />
    <ReduxForm
      onSubmit={[Function]} />
  </div>
</div>
`;
